cmd/determine-test-runners.rb

```diff
@@ -0,0 +1,206 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+require "formula"
+
+class Formula
+  def macos_only?
+    requirements.any? { |r| r.is_a?(MacOSRequirement) && !r.version_specified? }
+  end
+
+  def linux_only?
+    requirements.any?(LinuxRequirement)
+  end
+
+  def x86_64_only?
+    requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :x86_64) }
+  end
+
+  def arm64_only?
+    requirements.any? { |r| r.is_a?(ArchRequirement) && (r.arch == :arm64) }
+  end
+
+  def versioned_macos_requirement
+    requirements.find { |r| r.is_a?(MacOSRequirement) && r.version_specified? }
+  end
+
+  def compatible_with?(macos_version)
+    return true if versioned_macos_requirement.blank?
+
+    macos_version.public_send(versioned_macos_requirement.comparator, versioned_macos_requirement.version)
+  end
+
+  def dependents
+    @dependent_hash ||= {}
+    @dependent_hash[ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"].present?] ||= with_env(HOMEBREW_STDERR: "1") do
+      Utils.safe_popen_read(
+        HOMEBREW_BREW_FILE, "uses", "--formulae", "--eval-all", "--include-build", "--include-test", name
+      ).split("\n").map { |dependent| Formula[dependent] }.freeze
+    end
+
+    @dependent_hash[ENV["HOMEBREW_SIMULATE_MACOS_ON_LINUX"].present?]
+  end
+end
+
+module Homebrew
+  module_function
+
+  def determine_test_runners_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `determine-test-runners` <testing-formulae> [<deleted-formulae>]
```

Comment:
Sorry, I don't want to stall progress too hard here but: this is the sort of logic I feel uncomfortable being in Homebrew/homebrew-core where it can't really be tested. Ideally it'd be in `brew test-bot` or Homebrew/brew and have some tests written to exercise it.

Reply:
Nothing stops us from adding tests for this right here, though 🙂

I can add some, if you like.

Reply:
@carlocab None of the other `cmd/` in homebrew-core have meaningful tests. It's nicer for the tests and logic to live elsewhere (`test-bot` or `brew`) because they have unit tests suites, Sorbet, etc. setup.

Reply:
Ok, let me look at moving this to `test-bot` or `brew`.